### PR TITLE
Resync orders.

### PR DESCRIPTION
**Pull request by 0ffffh**
Remove redundant orders, and do not touch the rescheduled one

### CODE REVIEW

File: goods-partner-backend/src/main/java/com/goodspartner/facade/impl/DefaultDeliveryFacade.java [hunk 1]
```diff
@@ -79,4 +81,20 @@ public Delivery calculateDelivery(UUID deliveryId) {
...
     }
 
+    @Transactional
+    @Override
+    public Delivery sync(UUID deliveryId) {
+        orderFacade.removeRedundantOrdersBeforeSyncDelivery(deliveryId);
+        Delivery delivery = deliveryService.findByIdWithOrders(deliveryId);
+
+        if (!DRAFT.equals(delivery.getStatus())) {
+            throw new IllegalDeliveryStateForRecalculation();
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/facade/impl/DefaultDeliveryFacade.java:91]: Why you remove orders in two lines above if you suppose to throw exception just next line later

File: goods-partner-backend/src/main/java/com/goodspartner/repository/OrderExternalRepository.java [hunk 2]
```diff
@@ -72,4 +73,12 @@ List<OrderExternal> findAllByDeliveryAndCar(@Param("deliveryId") UUID deliveryId
     @Query("SELECT o FROM OrderExternal o WHERE o.id = :id")
     Optional<OrderExternal> findByIdWithDelivery(@Param("id") Long id);
 
+    @Modifying
+    @EntityGraph(attributePaths = {"delivery"})
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/repository/OrderExternalRepository.java:77]: EntityGraph ? What is the purpose here?

File: goods-partner-backend/src/main/java/com/goodspartner/facade/impl/DefaultDeliveryFacade.java [hunk 3]
```diff
@@ -79,4 +81,20 @@ public Delivery calculateDelivery(UUID deliveryId) {
         return delivery;
     }
 
+    @Transactional
+    @Override
+    public Delivery sync(UUID deliveryId) {
+        orderFacade.removeRedundantOrdersBeforeSyncDelivery(deliveryId);
+        Delivery delivery = deliveryService.findByIdWithOrders(deliveryId);
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/facade/impl/DefaultDeliveryFacade.java:88]: why do we need specifically withOrders?

File: goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultDeliveryService.java [hunk 4]
```diff
@@ -82,6 +82,13 @@ public Delivery findById(UUID deliveryId) {
                 .orElseThrow(() -> new DeliveryNotFoundException(deliveryId));
     }
 
+    @Transactional(readOnly = true)
+    @Override
+    public Delivery findByIdWithOrders(UUID deliveryId) {
+        return deliveryRepository.findByIdWithOrders(deliveryId)
+                .orElseThrow(() -> new DeliveryNotFoundException(deliveryId));
+    }
+
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultDeliveryService.java:91]: Not sure why we need it

**Review by belosh59 — CHANGES_REQUESTED** [review]

File: goods-partner-backend/src/main/java/com/goodspartner/facade/impl/DefaultDeliveryFacade.java (hunk 1 above)
Comment (0ffffh) [reply goods-partner-backend/src/main/java/com/goodspartner/facade/impl/DefaultDeliveryFacade.java]: fixed

File: goods-partner-backend/src/main/java/com/goodspartner/repository/OrderExternalRepository.java (hunk 2 above)
Comment (0ffffh) [reply goods-partner-backend/src/main/java/com/goodspartner/repository/OrderExternalRepository.java]: fixed

File: goods-partner-backend/src/main/java/com/goodspartner/facade/impl/DefaultDeliveryFacade.java (hunk 3 above)
Comment (0ffffh) [reply goods-partner-backend/src/main/java/com/goodspartner/facade/impl/DefaultDeliveryFacade.java]: fixed

File: goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultDeliveryService.java (hunk 4 above)
Comment (0ffffh) [reply goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultDeliveryService.java]: fixed

File: goods-partner-backend/src/main/java/com/goodspartner/web/controller/DeliveryController.java [hunk 5]
```diff
@@ -97,6 +97,18 @@ public DeliveryActionResponse approve(@ApiParam(value = "ID of Delivery to be ap
         return mapDeliveryActionResponse(deliveryFacade.approve(id, DeliveryAction.of(action)));
     }
 
+    @PostMapping("/{id}/sync")
+    @PreAuthorize("hasRole('ADMIN')")
```
[the remark this answers is not in the document]
Comment (0ffffh) [reply goods-partner-backend/src/main/java/com/goodspartner/web/controller/DeliveryController.java]: fixed

File: goods-partner-backend/src/test/resources/dbunit.yml [hunk 6]
```diff
@@ -1,4 +1,5 @@
 caseInsensitiveStrategy: !!com.github.database.rider.core.api.configuration.Orthography 'LOWERCASE'
 cacheConnection: false
 properties:
-  datatypeFactory: !!com.goodspartner.config.JsonBPostgresqlDataTypeFactory {}
\ No newline at end of file
+  datatypeFactory: !!com.goodspartner.config.JsonBPostgresqlDataTypeFactory { }
+  allowEmptyFields: true
```
[the remark this answers is not in the document]
Comment (0ffffh) [reply goods-partner-backend/src/test/resources/dbunit.yml:5]: this option allow dbrider to call insert with empty strings. In test dataset I have some null values

**Review by belosh59 — APPROVED** [review]

**Pull request merged**